src/sif/transitcost.cc

```diff
@@ -568,7 +568,18 @@ Cost TransitCost::EdgeCost(const baldr::DirectedEdge* edge,
                            const baldr::TransitDeparture* departure,
                            const uint32_t curr_time) const {
   // Separate wait time from time on transit
-  float wait_time = departure->departure_time() - curr_time;
+  // adjust the time if it is after midnight.
+  uint32_t current = curr_time;
+  while (current >= 86400)
+    current -= 86400;
+
+  float wait_time = departure->departure_time() - current;
+
+  // handle case where departure time is after midnight say 60 and
+  // current time is before midnight say 86340 which causes a negative
+  // wait time.
+  if (wait_time < 0.0f)
+    wait_time = (86400 - current) + departure->departure_time();
```

Comment:
this could simply be: if (wait_time < 0.0f) wait_time += 86400;
Also - would be good to put a constexpr in midgard/constants.h : e.g., constexpr uint32_t kSecondsPerDay = 86400;